api/src/evaluation/application/algorithm-simulator/index.js

```diff
@@ -0,0 +1,67 @@
+import Joi from 'joi';
+import { securityPreHandlers } from '../../../shared/application/security-pre-handlers.js';
+import { algorithmSimulatorController } from './algorithm-simulator-controller.js';
+import { identifiersType } from '../../../../lib/domain/types/identifiers-type.js';
+
+const register = async function (server) {
+  server.route([
+    {
+      method: 'POST',
+      path: '/api/admin/algorithm-simulator/smart-random',
+      config: {
+        pre: [
+          {
+            method: (request, h) =>
+              securityPreHandlers.hasAtLeastOneAccessOf([
+                securityPreHandlers.checkAdminMemberHasRoleSuperAdmin,
+                securityPreHandlers.checkAdminMemberHasRoleSupport,
+                securityPreHandlers.checkAdminMemberHasRoleMetier,
+                securityPreHandlers.checkAdminMemberHasRoleCertif,
+              ])(request, h),
+            assign: 'hasAuthorizationToAccessAdminScope',
+          },
+        ],
+        validate: {
+          payload: Joi.object({
+            data: {
+              attributes: {
+                'knowledge-elements': Joi.array()
+                  .items({
+                    source: Joi.string().required(),
+                    status: Joi.string().required(),
+                    'answer-id': identifiersType.answerId,
+                    'skill-id': Joi.string().required(),
+                  })
+                  .min(1)
+                  .required(),
+                answers: Joi.array().items({
+                  result: Joi.string().required(),
+                  'assessment-id': identifiersType.assessmentId,
+                  'challenge-id': identifiersType.challengeId,
+                  value: Joi.string().required(),
+                }),
+                skills: Joi.array().items({
+                  level: Joi.string().required(),
+                }),
+                challenges: Joi.array().items({
+                  id: Joi.string().required(),
+                }),
+                locale: Joi.string().required(),
```

Comment:
```suggestion
                locale: Joi.string().valid('fr', 'fr-fr', 'en', 'fr-be', 'nl').required(),
```

Voir s'il y a une constante avec toutes les locales dans le code.